src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -24,6 +24,140 @@ these dependencies for Axom.
     and experimentation?
   * Others?
 
+Determinism
+-----------
+
+We strive for as close to deterministic behavior in our builds as possible.
+By this, we mean that repeated builds should should act the same in the following
+regards:
+
+* Set of libraries with their options and versions
+* Compilers, compiler flags, and versions
+* Installed file and directory structure with permissions
+
+
+===========================================
+Build Scripts and Their Configuration Files
+===========================================
+
+There are three levels of build scripts or programs that drive TPL builds.
+As you move up the levels, and away from Spack, the scripts require less
+configuration and even build suites of builds at a time.
+
+Here is a brief description of what the levels are handling and what important
+configuration and input files they use, from lowest level to highest.
+
+Level 1: Spack
+--------------
+
+Spack is a multi-platform package manager that builds and installs multiple versions
```

Comment:
Should we add a link to spack's repo and/or docs here?

(nevermind, I see it at the end of the section)